logpy/facts.py

```diff
@@ -29,24 +31,41 @@ def add_fact(self, *inputs):
             self.index[key].add(fact)
 
     def __call__(self, *args):
-        def f(s):
-            args2 = reify(args, s)
+        """ Returns an evaluated (callable) goal, which returns a list of
+        substitutions which match args against a fact in the knowledge base.
```

Comment:
I'm still learning logic programming, so please let me know if these terms are incorrectly used. This docstring resulted from my attempts to understand what this method is doing and what the variable names meant.